src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/deleteroles/DeleteRoleAssignmentOrchestratorTest.java

```diff
@@ -130,6 +135,70 @@ void shouldDeleteRecordsFromRoleAssignment() throws Exception {
 
     }
 
+    @Test
+    @DisplayName("should delete any delete approved records that are present")
+    void shouldDeleteRecordsForApprovedItems() throws Exception {
+        mockRequest();
+        when(persistenceUtil.prepareHistoryEntityForPersistance(any(), any())).thenReturn(historyEntity);
+        assignmentRequest.getRequestedRoles().forEach(roleAssignment1 -> {
+            roleAssignment1.setStatus(DELETE_APPROVED);
+        });
+        mockHistoryEntity();
+
+        //set history entity into request entity
+        Set<HistoryEntity> historyEntities = new HashSet<>();
+        historyEntities.add(historyEntity);
+        requestEntity.setHistoryEntities(historyEntities);
+        sut.requestEntity = requestEntity;
```

Comment:
used setRequestEntity method .

Reply:
Refactored codebase to accommodate requested change.